apis/pkg/v1alpha1/function_types.go

```diff
@@ -40,8 +40,8 @@ type Function struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   FunctionSpec   `json:"spec"`
-	Status FunctionStatus `json:"status"`
+	Spec   FunctionSpec   `json:"spec,omitempty"`
+	Status FunctionStatus `json:"status,omitempty"`
 }
 
```

Comment:
FYI, down at line 93 you have `Endpoint` as a top-level field of `FunctionRevision`. I think that should be moved under `FunctionRevisionStatus`. I've done that in https://github.com/crossplane/crossplane/pull/4500 if you want to copy it.

Reply:
Moved the field to FunctionRevisionStatus.